### ISSUE TITLE
oxfmt moves oxlint-disable comments that re-enables oxlint warnings/errors

### ISSUE THREAD

**Issue by KuSh** [oxc-project/oxc#22062]
```ts
function x(foo: object) {
  return (
    // oxlint-disable-next-line no-unsafe-optional-chaining
    foo?.veryLongAttribute1?.veryLongAttribute2?.veryLongAttribute3
      ?.veryLongAttribute4 as any
  ).bar
}

function y(foo: object) {
  return (
    foo.veryLongAttribute1.veryLongAttribute2.veryLongAttribute3
      ?.veryLongAttribute4 as any // oxlint-disable-next-line no-unsafe-optional-chaining
  ).bar
}
```
is converted to

```ts
function x(foo: object) {
  return (
    // oxlint-disable-next-line no-unsafe-optional-chaining
    (
      foo?.veryLongAttribute1?.veryLongAttribute2?.veryLongAttribute3
        ?.veryLongAttribute4 as any
    ).bar
  );
}

function y(foo: object) {
  return (
    foo.veryLongAttribute1.veryLongAttribute2.veryLongAttribute3
      ?.veryLongAttribute4 as any
  ).bar; // oxlint-disable-next-line no-unsafe-optional-chaining
}
```

Prettier behaves the same but this reactivates oxlint errors. Perhaps disabling comments should be treated specifically?

Oxfmt [playground link](https://playground.oxc.rs/?t=formatter&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Atrue%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=function+x%28foo%3A+object%29+%7B%0A++return+%28%0A++++%2F%2F+oxlint-disable-next-line+no-unsafe-optional-chaining%0A++++foo%3F.veryLongAttribute1%3F.veryLongAttribute2%3F.veryLongAttribute3%0A++++++%3F.veryLongAttribute4+as+any%0A++%29.bar%0A%7D%0A%0Afunction+y%28foo%3A+object%29+%7B%0A++return+%28%0A++++foo.veryLongAttribute1.veryLongAttribute2.veryLongAttribute3%0A++++++%3F.veryLongAttribute4+as+any+%2F%2F+oxlint-disable-next-line+no-unsafe-optional-chaining%0A++%29.bar%0A%7D)

**Comment by AndrejBozin1** [comment]
oxfmt `0.51.0` (also reproduces on `0.50.0`) — a trailing `// oxlint-disable-line` placed after a block opener `{` gets relocated *inside* the block onto the next line, which breaks the suppression
   because the directive is no longer on the line it was authored against:

  ```tsx
  // Input
  const Foo = () => {
      try {
          const finding = JSON.parse('{}');
          return <JsonViewer value={finding} />;
      } catch (_e) { // oxlint-disable-line no-unused-vars
          return <LazyPage404 />;
      }
  };

  // Output (oxfmt 0.51.0)
  const Foo = () => {
      try {
          const finding = JSON.parse('{}');
          return <JsonViewer value={finding} />;
      } catch (_e) {
          // oxlint-disable-line no-unused-vars
          return <LazyPage404 />;
      }
  };
  ```

  After formatting, the `no-unused-vars` warning fires on `_e` again, because `oxlint-disable-line` only silences warnings on the line the comment is on — and the formatter has moved the comment off
  the `catch (_e) {` line.
